controllers/datadogagent/utils.go

```diff
@@ -1936,6 +1936,21 @@ func mergeAnnotationsLabels(logger logr.Logger, previousVal map[string]string, n
 	return mergedMap
 }
 
+func removeAnnotationLabel(vals map[string]string, valToRemove string) map[string]string {
```

Comment:
any reason not to just `delete(vals, valToRemove)`?

Reply:
I think I tried to copy the existing code for removing labels. I'll switch to `delete`